apps/admin/backend/src/exports/csv_results.ts

```diff
@@ -21,135 +22,295 @@ import { ScannerBatch } from '../types';
 import { Store } from '../store';
 import { tabulateElectionResults } from '../tabulation/full_results';
 
-function generateHeaders({
+/**
+ * Possible metadata attributes that can be included in the CSV, in the order
+ * the columns will appear in the CSV.
+ */
+const METADATA_ATTRIBUTES = [
+  'precinct',
+  'party',
+  'ballotStyle',
+  'votingMethod',
+  'scanner',
+  'batch',
+] as const;
```

Comment:
You may be thinking, or have thought before - we're reproducing and converting this list of attributes all over the place, maybe there's a refactor that we could do to reduce some of the boilerplate and even avoid bugs? 

And I think that's definitely true.

But, right now all of the filters or IDs are just strings, which would make generic typing easy (except that `votingMethod` is a restricted string). But that might not be the case for very long. We might need to describe geographical filters or groupings with additional level identifiers (e.g. "precinct" vs. "ward" vs. "county"). Or we might need to group voting method in multiple overlapping ways (e.g. in-person vs. mail or early voting in-person vs. election day in-person).

So, I'm punting on refactoring and am currently leaving a lot of this annoying boilerplate because I worry something more streamlined will restrict us in the near future.

Reply:
helpful context - thanks!